test/javascripts/turbolinks_test.coffee

```diff
@@ -98,113 +127,161 @@ describe 'Turbolinks', ->
-          assert.equal 1, ev.data.length
-          node = ev.data[0]
+    Turbolinks.visit "/some_request"
+    @server.respond()
 
-          assert.equal "turbo-area", node.id
-          assert.equal "turbo-area", node.getAttribute('refresh')
+    assert.equal true, @pushStateStub.called, 'Should call pushState instead of redirecting.'
 
-        @server.respondWith([200, { "Content-Type": "text/html" }, html_one]);
+  it 'inserts script with a new src from new doc into head on navigation', ->
+    @server.respondWith([200, { "Content-Type": "text/html" }, html_with_script_in_head])
 
-        Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area']}
-        @server.respond()
+    Turbolinks.visit "/some_request"
+    @server.respond()
 
-        $(document).off 'page:load'
+    assert.equal 1, document.querySelectorAll('script[src="merp"]').length, 'Should insert the script tag!'
 
-      it 'does not trigger the page:before-partial-replace event more than once', ->
-        handler = stub()
-        $(document).on 'page:before-partial-replace', ->
-          handler()
-          assert handler.calledOnce
+  it 'does not reinsert script with existing src from new doc into head on navigation', ->
+    @server.respondWith([200, { "Content-Type": "text/html" }, html_with_script_in_head])
 
-        @server.respondWith([200, { "Content-Type": "text/html" }, html_one]);
+    Turbolinks.visit "/some_request"
+    @server.respond()
 
-        Turbolinks.visit "/some_request", {partialReplace: true, onlyKeys: ['turbo-area']}
-        @server.respond()
+    Turbolinks.visit "/some_other_request"
+    @server.respond()
 
-        $(document).off 'page:before-partial-replace'
+    assert.equal true, document.querySelectorAll('script[src="merp"]').length == 1, 'Should insert the script tag once!'
 
-      it 'replaces and passes through the outermost nodes if a series of nodes got replaced', ->
-        currentBody = """
-          <div id="div1" refresh="div1">
-            <div id="div2" refresh-always>Refresh-always</div>
-          </div>
-        """
+  it 'inserts link tags with a new href from new doc into head on navigation', ->
+    @server.respondWith([200, { "Content-Type": "text/html" }, html_with_link_in_head])
 
-        $("body").append($(currentBody))
+    Turbolinks.visit "/some_request"
+    @server.respond()
 
-        $(document).on 'page:before-partial-replace', (ev) ->
-          nodes = ev.originalEvent.data
-          assert.equal 2, nodes.length
-          assert.equal 'div2', nodes[0].id
-          assert.equal 'div1', nodes[1].id
+    assert.equal 1, document.querySelectorAll('link[href="merp"]').length, 'Should add the new link tag!'
 
-        $(document).on 'page:load', (ev) ->
-          nodes = ev.originalEvent.data
-          assert.equal 1, nodes.length
-          assert.equal 'div1', nodes[0].id
+  it 'should remove link tags when navigating to a page with less of them', ->
```

Comment:
To explain further:
- Removing script tags does not stop them executing or remove the objects / listeners they created from memory
- Assuming that we removed them anyway, navigating away and navigating back would cause the script to be reevaluated
- Who knows what havoc this could wreak D: